tests/unittests/GraphGradTest.cpp

```diff
@@ -260,3 +260,30 @@ TEST(GraphAutoGrad, checkBatchedReduceAddGradTest) {
   EE.compile(CompilationMode::Train, TF);
   EE.compile(CompilationMode::Infer, F);
 }
+
+/// Check that we can differentiate functions that use GatherNode.
+TEST(GraphAutoGrad, checkGatherGradTest) {
```

Comment:
Can you add a case here with 2D indices (or 3D, etc.)? You don't have to create a new test, just add another node in the same `glow::Function`.